### PR TITLE
Algolia search for documentation

### PR DESCRIPTION
**Pull request by lluia**
algolia

Address #351 and continue the work started by @theevilhead in #381.

I couldn't install `docsearch.js` locally due to a certificate issue on my current machine, so I haven't verified the integration with **Algolia** works yet. 

## TODO

- [x] install and verify integration with **Algolia** works
- [ ] ~~test `<SearchBar />` using RTL~~
- [x] detect when `dropdown.js` fails to initialise and unmount component
- [x] fix layout issues on narrow viewports

### CODE REVIEW

**Comment by diegohaz** [comment]
We can remove the Spectrum link

**Comment by lluia** [comment]
@diegohaz 

I removed the **Spectrum** link as you suggested ( from the _footer_ too to be consistent 🤨 ).

I also removed the text next to the **Github** link:




Screenshot 2019-07-24 at 14 32 51

File: packages/website/src/components/Header.tsx [hunk 1]
```diff
@@ -211,7 +207,6 @@ export default function Header({ transparent }: HeaderProps) {
           <FaGithub style={{ fontSize: "1.2em" }} />
           <HiddenMediaQuery query="max-width: 768px">
             <Spacer width={8} />
-            GitHub
```
Comment (diegohaz) [inline packages/website/src/components/Header.tsx:215]: I think we can keep this and just adjust `768` to a value higher than that so it disappears before breaking the layout. Note that it includes the `isLarge` threshold as well.

File: packages/website/src/components/Footer.tsx [hunk 2]
```diff
@@ -80,14 +79,6 @@ export default function Footer() {
...
           <VisuallyHidden>Stackoverflow</VisuallyHidden>
         </Anchor>
-        <Anchor
-          href="https://spectrum.chat/reakit"
-          target="_blank"
-          onClick={track("reakit.footerSpectrumClick")}
-        >
-          <SpectrumLogo />
-          <VisuallyHidden>Spectrum</VisuallyHidden>
-        </Anchor>
```
Comment (diegohaz) [inline packages/website/src/components/Footer.tsx]: Let's keep the link on the Footer. There are other links there that aren't on header too.

**Comment by lluia** [comment]
@diegohaz I rescued this PR after I abandoned during some months 😆

I addressed your suggestions about the _breakpoint_ and bringing back the _Spectrum_ logo.

This is how it looks currently:



I made some adjustments 💅🏻 on the **dropdown UI**, this is how it looked originally:

Screenshot 2019-10-25 at 23 04 05

Everything was in `uppercase` so it was hard to have some visual hierarchy. I decided to `capitalize` the list items instead to differentiate them from the headers.

The **algolia logo** was cropped 🤷🏻‍♂️ , so I had to fix its height.

The layout of the list items was breaking sometimes 🧐

Screenshot 2019-10-25 at 23 09 53

and fixed it by giving them `width: 100%`. 

Apart from these issues I couldn't find other problems, have a look a let me know what you think 👍

**Comment by lluia** [comment]
@diegohaz do you think this PR is still relevant ☝️ ? Quite a few months have passed since I opened it, I'm happy to close if not 🙃

**Comment by diegohaz** [comment]
It's relevant for sure! I'll take a better look at this in the weekend! Thank you! :)

**Comment by lluia** [comment]
@diegohaz you think you'll find time to review this any time soon? 

I'm going to bring it up to date with `master`, I feel however it'll be good to move it forward as it's being open for a while 😇

**Comment by lluia** [comment]
@diegohaz updated this PR with `master` 👍🏻 , I clicked around the website locally and seems ok.

All the things above apply still, this is how it looks currently:



### Recap 
- we removed the **Spectrum** link on the header ( _as the layout was quite squeezed_ )
- we made a new `<SearchBar />` component to render this UI
- I made some adjustments on the **dropdown** to make it look prettier ( [explained here](https://github.com/reakit/reakit/pull/402#issuecomment-546526795) )

### Suggestions

I suggest to bring back the **Spectrum** link on the homepage header and just display the search bar on the `/docs` pages? ( _homepage seems a bit crowded with the search bar displayed and maybe is better to give visibility to Spectrum there?_ )

**Comment by lluia** [comment]
@diegohaz there's [some type errors](https://github.com/reakit/reakit/pull/402/files#diff-83020f36fc60cf702034bc2092cc7ed4) on `reakit-test-utils` package, not sure why 🤔 as I haven't touched it on this PR. 

It's been open for a long time, so I can always `cherry-pick` the commits in a new branch if that works for you 💆🏻‍♂️ 👍🏻

**Comment by diegohaz** [comment]
Hmm, it seems that yarn.lock has much more unrelated changes that may be causing the typescript errors. Maybe try to revert it to the version on master and install `docsearch.js` again?

**Comment by diegohaz** [comment]
Looks like it's passing now! 🎉

I think we can just ship it for now and iterate later on your suggestions.

Can you please push an empty commit just to trigger the Netlify build (I had deactivated it) so we can check if the build is working there?

**Pull request merged**